tests/libwnbd_tests/test.cpp

```diff
@@ -463,3 +463,39 @@ TEST(TestRead, Read2PBDisk) {
         (2LL << 50) / DefaultBlockSize,
         DefaultBlockSize);
 }
+
+void TestLiveResize(
+    uint64_t BlockCount = DefaultBlockCount,
+    uint32_t BlockSize = DefaultBlockSize,
+    uint64_t NewBlockCount = 2 * DefaultBlockCount)
+{
+    auto InstanceName = GetNewInstanceName();
+
+    MockWnbdDaemon WnbdDaemon(
+        InstanceName,
+        BlockCount,
+        BlockSize,
+        false,
+        false
+    );
+    WnbdDaemon.Start();
+
+    WNBD_CONNECTION_INFO ConnectionInfo = { 0 };
+    NTSTATUS Status = WnbdShow(InstanceName.c_str(), &ConnectionInfo);
+    ASSERT_FALSE(Status) << "couldn't retrieve WNBD disk info";
+
+    PWNBD_DISK WnbdDisk = WnbdDaemon.GetDisk();
+
+    ASSERT_EQ(BlockCount, WnbdDisk->Properties.BlockCount);
+
+    ASSERT_EQ(WnbdSetDiskSize(WnbdDisk, NewBlockCount), ERROR_SUCCESS);
+    ASSERT_EQ(NewBlockCount, WnbdDisk->Properties.BlockCount);
```

Comment:
We could add some assertions:

* when extending the disk, we should be able to write passed the old limit
* when shrinking the disk, we should get an error when trying to write passed the new limit